### DISCUSSION TITLE
menu-icon onClick handling

### DISCUSSION THREAD

**Discussion by rbureau85** [RocketCommunicationsInc/astro#213] [Q&A]
Hello Everyone,
Is there a action for click on the menu-icon by chance? I did not see an action for it in the storybook. 

Thanks

**Comment by micahjones13** [comment]
There isn't, but attaching an on click listener should work. Are you running into issues with that?

**Reply by rbureau85** [reply]
I have not tested this yet but wouldn't that make the entire GlobalHeader get triggered? I just added an RuxIcon withe an onclick and slot left-side. I guess that's as good of a solutions as any but then what is the menuIcon really. I just assumed it had a the purpose of being an menu. If that makes sense.

**Reply by micahjones13** [reply]
Using [event.stopPropagation()](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation) on the icon's event should prevent any gsb events from triggering. The menu-icon prop on gsb is meant to be a shortcut for allowing any rux-icon into gsb, not specifically for the purpose of a menu. Using slots like you have with the `<rux-icon slot="left-side">` should allow you to handle events on the slotted component with relative ease.
